.github/workflows/maintenance.yaml

```diff
@@ -0,0 +1,29 @@
+on:
+  push:
+    branches: [ master ]
+  schedule:
+    - cron: '*/5 * * * *'
+  workflow_dispatch: # Enables on-demand/manual triggering
+jobs:
+  job:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+      - run: |
+          go get github.com/aws/aws-sdk-go
+          go mod tidy
+          go generate ./...
+          cd tools/provider-schema
+          terraform init -upgrade
+          terraform providers schema -json > schema.json
```

Comment:
Ah, I'm surprised that ubuntu-latest includes Terraform by default.
https://github.com/actions/virtual-environments/blob/cd453927cac82e4dc2658bb65d42b2c97eb83212/images/linux/scripts/installers/terraform.sh

However, this action may fail when a new version is released. It seems good to pin the version using setup-terraform.
https://github.com/hashicorp/setup-terraform

Reply:
Thanks for the pointer, considered installing `tfenv` as [recommended in our docs](https://github.com/terraform-linters/tflint-ruleset-aws/tree/master/tools/provider-schema) but knew there were issues installing it when terraform was already installed. I'll look into that instead